x-pack/plugins/ml/public/application/timeseriesexplorer/timeseriesexplorer.js

```diff
@@ -1044,15 +1014,13 @@ export class TimeSeriesExplorer extends React.Component {
           selectedEntities={this.props.selectedEntities}
           bounds={bounds}
         >
-          {functionDescription && (
-            <PlotByFunctionControls
-              selectedJobId={selectedJobId}
-              selectedDetectorIndex={selectedDetectorIndex}
-              selectedEntities={this.props.selectedEntities}
-              functionDescription={functionDescription}
-              setFunctionDescription={this.setFunctionDescription}
-            />
-          )}
+          <PlotByFunctionControls
```

Comment:
It was correct not to render the component when `functionDescription ` is not defined.  Now you have multiple checks inside, and anyway return `null` when it's undefined. So would be better to keep the check outside.

Reply:
It's necessary to call this PlotByFunctionControls because it does some some fetching to get the highest scoring record and then picks a function by default. Before, we can optionally not render the component because the data handling was done in the `TimeSeriesExplorer` level. With the latest changes, it's necessary that the effect is called.

One thing wen can do is to not render it when the detector function is metric.

Reply:
yes, I missed  the `setFunctionDescription ` part 👍